src/containerapp/azext_containerapp/custom.py

```diff
@@ -3656,6 +3660,72 @@ def disable_dapr(cmd, name, resource_group_name, no_wait=False):
     except Exception as e:
         handle_raw_exception(e)
 
+def init_dapr_component(cmd, resource_group_name, environment_name):
+    _validate_subscription_registered(cmd, CONTAINER_APPS_RP)
+
+    # Create the Redis service
+    redis_capp_name = DAPR_REDIS_PREFIX +'-' + ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(5))
+    logger.info("Creating Redis service '%s'...", redis_capp_name)
+    try:
+        redis_capp_def = create_redis_service(cmd, redis_capp_name, environment_name, resource_group_name)
+    except Exception as e:
+        raise ValidationError("Failed to create Redis service.") from e
+    
+    if redis_capp_def is None:
+        raise ValidationError("Failed to create Redis service, did not receive a response.")
+
+    # Look up the Redis secret
+    _get_existing_secrets(cmd, resource_group_name, redis_capp_name, redis_capp_def)
+    redis_secrets = safe_get(redis_capp_def, "properties", "configuration", "secrets", default=[])
+    if len(redis_secrets) == 0:
+        raise ValidationError("Failed to setup Dapr components, no secrets found for Redis service.")
+    
+    redis_secret_value = None
+    for secret in redis_secrets:
+        if safe_get(secret, "name") == DAPR_REDIS_SECRET_NAME:
+            redis_secret_value = safe_get(secret, "value")
+            break
+
+    if redis_secret_value is None:
+        raise ValidationError(f"Failed to setup Dapr components, {DAPR_REDIS_SECRET_NAME} secret not found for Redis service.")
+    
+    # The Redis secret is in the format:
+    # requirepass mypassword\ndir /mnt/data\nport 6379\nprotected-mode yes\nappendonly yes\n
+    redis_password = None
+    for line in redis_secret_value.splitlines():
+        if line.startswith(DAPR_REDIS_CONFIG_PASSWORD_KEY):
+            redis_password = line.split(" ")[1]
+            break
+
+    if redis_password is None:
+        raise ValidationError(f"Failed to setup Dapr components, {DAPR_REDIS_SECRET_NAME} secret does not contain a password for Redis service.")
+    
+    # Create the Dapr components
+    redis_host = f"{redis_capp_name}:{DAPR_REDIS_PORT}"
```

Comment:
I think technically the port should be read from the `exposedPort` in `redis_capp_def` although that's unlikely to be different.

Reply:
Yes, I initially wrote it but then it's unlikely to change. BUT, now that redis can *pre-exist*, let me add it. This will allow users to create their own Redis with the name `dapr-redis` and continue to use it here.